components/common-go/baseserver/server.go

```diff
@@ -31,28 +31,57 @@ func New(name string, opts ...Option) (*Server, error) {
 		cfg:  cfg,
 	}
 
-	if initErr := server.initializeHTTP(); initErr != nil {
-		return nil, fmt.Errorf("failed to initialize http server: %w", initErr)
+	if initErr := server.initializeDebug(); initErr != nil {
+		return nil, fmt.Errorf("failed to initialize debug server: %w", initErr)
 	}
-	if initErr := server.initializeGRPC(); initErr != nil {
-		return nil, fmt.Errorf("failed to initialize grpc server: %w", initErr)
+
+	if server.isHTTPEnabled() {
+		if initErr := server.initializeHTTP(); initErr != nil {
```

Comment:
nit: [no side effects in comparisons](https://github.com/gitpod-io/go-style-guide/blob/master/style.md#no-side-effects-in-if-statements)

```suggestion
		err := server.initializeHTTP()
		if err != nil {
```

Reply:
Interestingly enough, gometalinter flags this for style and suggests inlining the error check. There's a benefit to not assigning the error - you can continue to re-use `err` as the variable name (if you've not assigned it wider scope) without shadowing the error, it's quite common in the standard library.

I'll fix it here, I don't really mind but this feels like a grey area for the `side effects in comparison` area as here it feels like less of a side-effect and more a direct effect itself. For me a side-effect here would to touch something unrelated to HTTP. Either way, I'll take a closer look at the guide and try to clarify my thoughts on this with a PR against the guide, we can discuss there :)